pkg/dns/config/sync_api.go

```diff
@@ -0,0 +1,120 @@
+/*
```

Comment:
can we name this sync_configmap.go? sync_api would refer to "the sync API" in my mind